cmake/iDynTreeDependencies.cmake

```diff
@@ -51,8 +51,20 @@ if(IDYNTREE_USES_YARP AND YARP_FOUND)
   endif()
 endif()
 
+find_library(WORHP_LIB worhp HINTS ENV WORHP_DIR)
+option(IDYNTREE_USES_WORHP "Use WORHP as a solver in optimalcontrol." WORHP_LIB)
+if (IDYNTREE_USES_WORHP)
+    if (NOT WORHP_LIB)
+        message(FATAL_ERROR "The IDYNTREE_USES_WORHP option was selected, but it was not possible to find WORHP.")
+    endif()
+
+    find_path(WORHP_INCLUDE_DIRS worhp HINTS ENV WORHP_DIR)
+
+endif()
```

Comment:
Could you please refactor this logic in a `FindWORHP.cmake` so that we could just have the usual line: 
~~~
idyntree_handle_dependency(WORHP)
~~~

If I am not wrong, I guess this `FindWORHP.cmake` file could be as simple as: 
~~~


include(FindPackageHandleStandardArgs)

find_path(WORHP_INCLUDE_DIR worhp HINTS ENV WORHP_DIR)
mark_as_advanced(WORHP_INCLUDE_DIR)
find_library(WORHP_LIBRARY worhp HINTS ENV WORHP_DIR)
mark_as_advanced(WORHP_LIBRARY)

find_package_handle_standard_args(WORHP DEFAULT_MSG WORHP_INCLUDE_DIR WORHP_LIBRARY)

if(WORHP_FOUND)
    set(PCAP_INCLUDE_DIRS ${PCAP_INCLUDE_DIR})
    set(PCAP_LIBRARIES ${PCAP_LIBRARY})

    if(NOT TARGET PCAP::PCAP)
      add_library(WORHP::WORHP UNKNOWN IMPORTED)
      set_target_properties(WORHP::WORHP PROPERTIES
        INTERFACE_INCLUDE_DIRECTORIES "${WORHP_INCLUDE_DIR}")
      set_property(TARGET WORHP::WORHP APPEND PROPERTY
        IMPORTED_LOCATION "${WORHP_LIBRARY}")
    endif()
endif()
~~~

For an example of this simple `Find***.cmake` scripts, check: 
* https://github.com/Kitware/CMake/blob/master/Modules/FindZLIB.cmake
* https://github.com/Kitware/CMake/blob/master/Modules/FindJPEG.cmake
Even if those one are more complex as they handle multiple config generators (such as visual studio) and they also handled the version.

Reply:
I guess that ``PCAP`` is a leftover. Is it?